comp/forwarder/defaultforwarder/resolver/domain_resolver.go

```diff
@@ -435,3 +435,15 @@ func (r *domainResolver) GetAuthorizers() (res []authHeader) {
 func (r *domainResolver) GetConfigName() string {
 	return r.configName
 }
+
+func (r *domainResolver) IsFQDN() bool {
+	return strings.HasSuffix(r.GetBaseDomain(), ".")
+}
+
+func (r *domainResolver) ToFQDN() {
+	r.SetBaseDomain(r.GetBaseDomain() + ".")
+}
```

Comment:
BaseDomain should not be modified when the forwarder is running. It is used to locate the correct worker queue, and would cause errors. (`SetBaseDomain` is an unfortunate wart, it shouldn't really exist at all).